src/framework/pagination/Pagination.jsx

```diff
@@ -0,0 +1,108 @@
+import React, {
+  Component,
+  PropTypes,
+} from 'react';
+
+const propTypes = {
+  currentPage: PropTypes.number,
+  onChangePage: PropTypes.func.isRequired,
+  totalPages: PropTypes.number,
+  visiblePages: PropTypes.number,
+};
+
+const defaultProps = {
+  currentPage: 0,
+  totalPages: 100,
+  visiblePages: 7,
+};
+
+class Pagination extends Component {
+  getPageList() {
+    const { visiblePages, currentPage, totalPages } = this.props;
+
+    const pagesToDisplay = Math.min(visiblePages, totalPages);
+
+    const till = this.validatePage(
+      currentPage + Math.floor(pagesToDisplay / 2)
+    ) + 1;
+
+    const from = this.validatePage(
+      till - pagesToDisplay
+    );
+
+    const result = [...Array(pagesToDisplay).keys()].map(x => x + from);
+
+    return result;
+  }
+
+  validatePage(currentPage, totalPages = this.props.totalPages) {
```

Comment:
hum... it took me a while to figure what we are doing here.... I guess we are restricting any potential page out of the boundaries to go through.... what about something which says what we are exactly doing.... `setBoundaries` or something?

Reply:
or i don't know....`setBoundaries` sounds also a bit weird... @arijitdasgupta what do you think?

Reply:
What about this one?

Reply:
`limitPageIndex` as agreed